add results section for scheduler experiments (#21)
sayakpaul/dreambooth-keras: README.md

Follow [this guide](https://huggingface.co/docs/diffusers/main/en/using-diffusers/kerascv) to know more. 

### Experimental results through various scheduler settings:

<details>
<summary>
We have converted fine-tuned checkpoint for the dog images into Diffusers compatible StableDiffusionPipeline and ran various experiments with different scheduler settings. For example, the following parameters of the `DDIMScheduler` are tested on a different set of `guidance_scale` and `num_inference_steps`.

> </summary>

Reviewer: The code quotes are not rendering:

<img width="841" alt="image" src="https://user-images.githubusercontent.com/22957388/217406174-c782e0d7-a0c8-4523-948e-41e69b84ef88.png">

```python
num_inference_steps_list = [25, 50, 75, 100]
guidance_scale_list = [7.5, 15, 30]

scheduler_configs = {
  "DDIMScheduler": {
      "beta_value": [
          [0.000001, 0.02], 
          [0.000005, 0.02], 
          [0.00001, 0.02], 
          [0.00005, 0.02], 
          [0.0001, 0.02], 
          [0.0005, 0.02]
      ],
      "beta_schedule": [
          "linear", 
          "scaled_linear", 
          "squaredcos_cap_v2"
      ],
      "clip_sample": [True, False],
      "set_alpha_to_one": [True, False],
      "prediction_type": [
          "epsilon", 
          "sample", 
          "v_prediction"
      ]
  }
}
```

> Below shows the comparison between different values of `beta_schedule` parameters while others fixed as default. Take a look at [the original report](https://docs.google.com/spreadsheets/d/1_NhWuORn5ByEnvD9T3X4sHUnz_GR8uEtbE5HbI98hOM/edit?usp=sharing) which includes the results from other schedulers such as `PNDMScheduler` and `LMSDiscrete`. 

Reviewer: ```suggestion
Below is the comparison between different values of `beta_schedule` parameters while others are fixed to their default values. Take a look at [the original report](https://docs.google.com/spreadsheets/d/1_NhWuORn5ByEnvD9T3X4sHUnz_GR8uEtbE5HbI98hOM/edit?usp=sharing) which includes the results from other schedulers such as `PNDMScheduler` and `LMSDiscreteScheduler`. 
```

> It is often observed the default settings doesn't guarantee to generate better quality images. For example, the default values `guidance_scale` and `beta_schedule` are set to 7.5 and `linear`. However, when `guidance_scale` is set to 7.5, `scaled_linear` of the `beta_schedule` seems to work better. Or, when `beta_schedule` is set to `linear`, higher `guidance_scale` seems to work better. 

Reviewer: ```suggestion
It is often observed the default settings do guarantee to generate better quality images. For example, the default values of `guidance_scale` and `beta_schedule` are set to 7.5 and `linear`. However, when `guidance_scale` is set to 7.5, `scaled_linear` of the `beta_schedule` seems to work better. Or, when `beta_schedule` is set to `linear`, higher `guidance_scale` seems to work better. 
```

![](https://i.postimg.cc/QsW-CKTcv/DDIMScheduler.png)

> We have also run 4,800 experiments which generated 38,400 images in total. Those experiments are logged in Weights and Biases, and they are open to public. If you are curious, do check them out [here](https://wandb.ai/chansung18/SD-Scheduler-Explore?workspace=user-chansung18) as well as the [script](https://gist.github.com/deep-diver/0a2deb2cd369ab8c1bf3ee12f47d272a) that run the experiments. 

Reviewer: ```suggestion
We ran 4,800 experiments which generated 38,400 images in total. Those experiments are logged in Weights and Biases. If you are curious, do check them out [here](https://wandb.ai/chansung18/SD-Scheduler-Explore?workspace=user-chansung18) as well as the [script](https://gist.github.com/deep-diver/0a2deb2cd369ab8c1bf3ee12f47d272a) that was used to run the experiments. 
```
